components/cam/src/dynamics/se/dp_coupling.F90

```diff
@@ -1,543 +1,623 @@
+           call pbuf_get_field(pbuf_chnk, frontgf_idx, pbuf_frontgf)
+           call pbuf_get_field(pbuf_chnk, frontga_idx, pbuf_frontga)
+        end if
+        call block_to_chunk_recv_pters(lchnk,pcols,pver+1,tsize,cpter)
+        do icol = 1,ncols
+          phys_state(lchnk)%ps  (icol) = cbuffer(cpter(icol,0))
+          phys_state(lchnk)%phis(icol) = cbuffer(cpter(icol,0)+1)
+          do ilyr = 1,pver
+            phys_state(lchnk)%t    (icol,ilyr) = cbuffer(cpter(icol,ilyr))
+            phys_state(lchnk)%u    (icol,ilyr) = cbuffer(cpter(icol,ilyr)+1)
+            phys_state(lchnk)%v    (icol,ilyr) = cbuffer(cpter(icol,ilyr)+2)
+            phys_state(lchnk)%omega(icol,ilyr) = cbuffer(cpter(icol,ilyr)+3)
+             if (use_gw_front) then
+                pbuf_frontgf(icol,ilyr) = cbuffer(cpter(icol,ilyr)+4)
+                pbuf_frontga(icol,ilyr) = cbuffer(cpter(icol,ilyr)+5)
+             end if
+             do m = 1,pcnst
+                phys_state(lchnk)%q(icol,ilyr,m) = cbuffer(cpter(icol,ilyr)+tsize-pcnst-1+m)
+             end do ! m
+          end do ! ilyr
+        end do ! icol
+      end do ! lchnk
+
+      deallocate( bbuffer )
+      deallocate( cbuffer )
+      deallocate( bpter )
+
+    end if ! local_dp_map
     call t_stopf('dpcopy')
 
+    ! Deallocate the temporary arrays
+    deallocate(ps_tmp)
+    deallocate(zs_tmp)
+    deallocate(T_tmp)
+    deallocate(uv_tmp)
+    deallocate(q_tmp)
+    deallocate(w_tmp)
+
     call t_startf('derived_phys')
     call derived_phys(phys_state,phys_tend,pbuf2d)
     call t_stopf('derived_phys')
 
 !for theta there is no need to multiply omega_p by p
 #ifndef MODEL_THETA_L
-!$omp parallel do private (lchnk, ncols, ilyr, icol)
-    do lchnk=begchunk,endchunk
-      ncols=get_ncols_p(lchnk)
-      do ilyr=1,pver
-        do icol=1,ncols
-          if (.not. single_column) then
-          phys_state(lchnk)%omega(icol,ilyr)=phys_state(lchnk)%omega(icol,ilyr)*phys_state(lchnk)%pmid(icol,ilyr)
-          endif
-        end do
-      end do
-   end do
+    !$omp parallel do private (lchnk, ncols, ilyr, icol)
+    do lchnk = begchunk,endchunk
+      ncols = get_ncols_p(lchnk)
+      do ilyr = 1,pver
+        do icol = 1,ncols
+          if (.not.single_column) then
+            phys_state(lchnk)%omega(icol,ilyr) = phys_state(lchnk)%omega(icol,ilyr) \
+                                                *phys_state(lchnk)%pmid(icol,ilyr)
+          end if
+        end do ! icol
+      end do ! ilyr
+    end do ! lchnk
 #endif
 
-   if (write_inithist() ) then
-      do lchnk=begchunk,endchunk
-         call outfld('T&IC',phys_state(lchnk)%t,pcols,lchnk)
-         call outfld('U&IC',phys_state(lchnk)%u,pcols,lchnk)
-         call outfld('V&IC',phys_state(lchnk)%v,pcols,lchnk)
-         call outfld('PS&IC',phys_state(lchnk)%ps,pcols,lchnk)
-         do m=1,pcnst
+    if ( write_inithist() ) then
+      if (fv_nphys > 0) then
+
+        ncol_d = np*np
+        ! Allocate temporary arrays
```

Comment:
why are there so many commented out lines? if you still need them, could you insure a comment on why?

Reply:
I should delete these. There was some complications when trying to get the init files working properly, so I had many different versions. Now that I've verified that it works correctly I can delete these.

Reply:
resolving this comment as these lines were removed.